From 70db3626abb435555a211e8b24df2507217d102b Mon Sep 17 00:00:00 2001
From: Iceows <mounierr07@gmail.com>
Date: Tue, 12 Jul 2022 10:33:46 +0200
Subject: [PATCH] Fixed errors on the ModermMediaScanner (Huawei)

This patch prevents the media scanner from crashing when starting the phone. This problem prevented displaying the default sounds (alarm, ringtones..). The root cause of the error is a null volume name for system
---
 .../media/scan/ModernMediaScanner.java          | 17 ++++++++++++++---
 1 file changed, 14 insertions(+), 3 deletions(-)

diff --git a/src/com/android/providers/media/scan/ModernMediaScanner.java b/src/com/android/providers/media/scan/ModernMediaScanner.java
index 8927bfa5..885bacae 100644
--- a/src/com/android/providers/media/scan/ModernMediaScanner.java
+++ b/src/com/android/providers/media/scan/ModernMediaScanner.java
@@ -367,11 +367,22 @@ public class ModernMediaScanner implements MediaScanner {
             mRoot = root;
             mReason = reason;
 
+            MediaVolume VolumeTmp=null;
             if (FileUtils.contains(Environment.getStorageDirectory(), root)) {
-                mVolume = MediaVolume.fromStorageVolume(FileUtils.getStorageVolume(mContext, root));
-            } else {
-                mVolume = MediaVolume.fromInternal();
+                try {
+                    VolumeTmp = MediaVolume.fromStorageVolume(FileUtils.getStorageVolume(mContext, root));
+                }
+                catch (FileNotFoundException e)
+                {
+                    Log.w(TAG,"Can't find volume for " + root);
+                }
             }
+
+            if (VolumeTmp == null)
+                mVolume = MediaVolume.fromInternal();
+            else
+                mVolume=VolumeTmp;
+            
             mVolumeName = mVolume.getName();
             mFilesUri = MediaStore.Files.getContentUri(mVolumeName);
             mSignal = new CancellationSignal();
-- 
2.33.0.windows.2

